python/src/scipp/plot/resampling_model.py

```diff
@@ -0,0 +1,198 @@
+            else:
+                plan.insert(0, (coord, edge))
+        for coord, edge in plan:
+            try:
+                var = sc.rebin(var, dim, coord, edge)
+            except RuntimeError:  # Limitation of rebin for slice of inner dim
+                var = sc.rebin(var.copy(), edge.dims[-1], coord, edge)
+        return var
+
+    def _make_edges(self, params):
+        edges = []
+        for dim, par in params.items():
+            if isinstance(par, int):
+                continue
+            low, high, unit, res = par
+            edges.append(
+                sc.Variable(dims=[dim],
+                            unit=unit,
+                            values=np.linspace(low, high, num=res + 1)))
+        return edges
+
+    def _call_resample(self):
+        out = self._array
+        params = {}
+        for dim, s in self.bounds.items():
+            dim = str(dim)
+            if s is None:
+                coord = self._array.meta[dim]
+                low = coord.values[0]
+                high = coord.values[-1]
+                params[dim] = (low, high, coord.unit, self.resolution[dim])
+            elif isinstance(s, int):
+                out = out[dim, s]
+                params[dim] = s
+            else:
+                low, high = s
+                if isinstance(low, int):
+                    out = out[dim, low:high]
+                    low = out.meta[dim][dim, 0]
+                    high = out.meta[dim][dim, -1]
+                else:
+                    # cannot pre-select bins for multi-dim coords
+                    if len(out.meta[dim].dims) == 1:
+                        out = out[sc.get_slice_params(out.data, out.meta[dim],
+                                                      low, high)]
+                params[dim] = (low.value, high.value, low.unit,
+                               self.resolution[dim])
+        if params == self._home_params:
+            # This is a crude caching mechanism for past views. Currently we
+            # have the "back" buttons disabled in the matplotlib toolbar, so
+            # we cache only the "home" view. This is the most expensive to
+            # create anyway.
+            self._resampled_params = self._home_params
+            self._resampled = self._home
+        elif self._resampled is None or params != self._resampled_params:
+            self._resampled_params = params
+            self._edges = self._make_edges(params)
+            self._resampled = self._resample(out)
+        if self._home is None:
+            self._home = self._resampled
+            self._home_params = self._resampled_params
+
+
+class ResamplingBinnedModel(ResamplingModel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # TODO See #1469. This is a temporary hack to work around the
+        # conversion of coords to edges in model.py.
+        for name in self._array.meta:
+            if name == 'spectrum':
```

Comment:
Is this only applicable to `spectrum` or is it a fix for neutron data?
If it is only for spectrum, then you could have `if "spectrum" in self._array.meta:`.
If it isn't just for spectrum, then something like `if not bin_edge`:`